velox/functions/lib/Re2Functions.cpp

```diff
@@ -1001,30 +1012,42 @@ std::pair<PatternKind, vector_size_t> determinePatternKind(StringView pattern) {
              (patternStr[i] != '%' && patternStr[i] != '_')) {
         i++;
       }
+      fixedPatternEnd = i - 1;
     }
   }
 
   // Pattern contains wildcard characters only.
   if (fixedPatternStart == -1) {
     if (!anyCharacterWildcardCount) {
-      return {PatternKind::kExactlyN, singleCharacterWildcardCount};
+      return PatternMetadata{
+          PatternKind::kExactlyN, singleCharacterWildcardCount};
     }
-    return {PatternKind::kAtLeastN, singleCharacterWildcardCount};
+    return PatternMetadata{
+        PatternKind::kAtLeastN, singleCharacterWildcardCount};
   }
   // Pattern contains no wildcard characters (is a fixed pattern).
   if (wildcardStart == -1) {
-    return {PatternKind::kFixed, patternLength};
+    return PatternMetadata{PatternKind::kFixed, patternLength};
   }
   // Pattern is generic if it has '_' wildcard characters and a fixed pattern.
   if (singleCharacterWildcardCount) {
-    return {PatternKind::kGeneric, 0};
+    return PatternMetadata{PatternKind::kGeneric, 0};
   }
-  // Classify pattern as prefix pattern or suffix pattern based on the
-  // positions of the fixed pattern and contiguous wildcard character stream.
+  // Classify pattern as prefix, fixed center, or suffix pattern based on the
+  // position and count of the wildcard character sequence and fixed pattern.
   if (fixedPatternStart < wildcardStart) {
-    return {PatternKind::kPrefix, wildcardStart};
+    return PatternMetadata{PatternKind::kPrefix, wildcardStart};
+  }
+  if (numWildcardSequences > 1) {
```

Comment:
// if numWildcardSequences>1, then fixed pattern must be in between them
btw in this case numWildcardSequences should be exactly two right?
shall we make it numWildcardSequences=2?